src/layout/FormFields.component.js

```diff
@@ -236,10 +236,19 @@ function createAvatarEditor(fieldBase, d2, valueStore) {
     })
 }
 
-function createVerifyButton(fieldBase, valueStore) {
+function createModalField({ fieldBase, valueStore, onUpdate, d2 }) {
     return Object.assign({}, fieldBase, {
-        component: VerifyEmail,
-        props: { userEmail: valueStore.state['email'] || '' },
+        component: ModalField,
+        props: {
+            onUpdate,
+            userEmail: valueStore.state['email'] || '',
+            userEmailVerified: d2?.currentUser?.emailVerified,
+            setUserEmail: (email) => {
+                valueStore.state['email'] = email
+                valueStore.state['emailUpdated'] = true
+                valueStore.setState(valueStore.state)
```

Comment:
initially, we'd done `valueStore.setState({...valueStore.state, email})` (or similar), but then this did not actually set the new email state 🤔 (I think there's some nuances to component setState that I no longer know)